### PR TITLE
Update ApiControllerAttribute.xml

### LINKED ISSUE #103: Elaborate what ApiControllerAttribute does
The documentation says it is used for "improving the developer experience".

https://github.com/aspnet/ApiDocs/blob/7a260832bf46556b49d59e0de86a1d36663b2f80/aspnet-core/xml/Microsoft.AspNetCore.Mvc/ApiControllerAttribute.xml#L32-L33

It would probably be sufficient to link back to https://docs.microsoft.com/en-us/aspnet/core/web-api/?view=aspnetcore-2.2 via hyperlinking the text "improving the developer experience".  When I first read this phrase, I assumed it meant RAD tooling within Visual Studio meant to upsell Visual Studio as a great tool.

According to https://docs.microsoft.com/en-us/aspnet/core/web-api/?view=aspnetcore-2.2 , it appears that ApiController does the following:

* **Automatic HTTP 400 responses**: Model validation errors automatically trigger an HTTP 400 response.
* **Binding source parameter inference**: Without the `[ApiController]` attribute, binding source attributes are explicitly defined. Without `[ApiController]` or other binding source attributes like `[FromQuery]`, the ASP.NET Core runtime attempts to use the complex object model binder. The complex object model binder pulls data from value providers (which have a defined order). For instance, the 'body model binder' is always opt in.
* **Multipart/form-data request inference**: When an action parameter is annotated with the [FromForm] attribute, the multipart/form-data request content type is inferred.
* **Attribute routing requirement**: Attribute routing becomes a requirement
* **Problem details responses for error status codes**:
    > In ASP.NET Core 2.2 or later, MVC transforms an error result (a result with status code 400 or higher) to a result with ProblemDetails. ProblemDetails is:
    > 
    > * A type based on the RFC 7807 specification.
    > * A standardized format for specifying machine-readable error details in an HTTP response.

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #103

### CODE REVIEW

File: aspnet-core/xml/Microsoft.AspNetCore.Mvc/ApiControllerAttribute.xml [hunk 1]
```diff
@@ -55,7 +55,7 @@
             </para><para>
             When decorated on an assembly, all controllers in the assembly will be treated as controllers with API behavior.
```
Comment (Rick-Anderson) [inline aspnet-core/xml/Microsoft.AspNetCore.Mvc/ApiControllerAttribute.xml:56]:
```suggestion
            When decorated on an assembly, all controllers in the assembly are treated as controllers with API behavior.
```

**Comment by Rick-Anderson** [comment]
@jzabroski please review and thanks for bringing this to our attention. Sorry it has taken so long to respond.

File: aspnet-core/xml/Microsoft.AspNetCore.Mvc/ApiControllerAttribute.xml [hunk 2]
```diff
@@ -53,9 +53,9 @@
             Controllers decorated with this attribute are configured with features and behavior targeted at improving the
             developer experience for building APIs.
             </para><para>
-            When decorated on an assembly, all controllers in the assembly will be treated as controllers with API behavior.
+            When decorated on an assembly, all controllers in the assembly are treated as controllers with API behavior.
             </para></summary>
-    <remarks>To be added.</remarks>
+    <remarks>For more information, see <see href="https://docs.microsoft.com/en-us/aspnet/core/web-api/#apicontroller-attribute">Cookie policy middleware</see>.</remarks>
```
Comment (JamesNK) [inline aspnet-core/xml/Microsoft.AspNetCore.Mvc/ApiControllerAttribute.xml:58]:
```suggestion
    <remarks>For more information, see <see href="https://docs.microsoft.com/aspnet/core/web-api/#apicontroller-attribute">Cookie policy middleware</see>.</remarks>
```

**Review by JamesNK — APPROVED** [review]
Do changes like this need to be mirrored in the aspnetcore repo? If they aren't, won't they be overwritten next time they're regenerated from C# XML comments?

**Comment by Rick-Anderson** [comment]
> Do changes like this need to be mirrored in the aspnetcore repo? If they aren't, won't they be overwritten next time they're regenerated from C# XML comments?

Is that what you and @wtgodbe did in https://github.com/dotnet/AspNetApiDocs/pull/187/files ?

**Comment by jzabroski** [comment]
> @jzabroski please review and thanks for bringing this to our attention. Sorry it has taken so long to respond.
> -- @Rick-Anderson 

This would not be the longest open issue I have with Microsoft, so no worries. :) Heck, I've had PRs open for 2-3 years before someone finally chose to merge them. :)

I don't understand why this behavior is a "Cookie middleware policy"?

Separately, the context around my issue at the time was I found the `ProblemDetails` abstraction to be rather confusing/limiting, and was mildly amused by the "improve developer productivity" remark.  In particular, this gotcha was super annoying:

> "The round-tripping behavior for Extensions is determined by the implementation of the Input \ Output formatters. In particular, complex types or collection types may not round-trip to the original type when using the built-in JSON or XML formatters."
> -- https://docs.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.problemdetails.extensions?view=aspnetcore-2.2#Microsoft_AspNetCore_Mvc_ProblemDetails_Extensions

Sorry that I do not have more feedback than this.  I jump around doing all kinds of projects so I last worked on this type of code base in February 2019 (when I opened #103) according to my contemporaneous notes, and that was all I wrote down.  Looking at [RFC 7807 Section 3.2 Extension Members](https://tools.ietf.org/html/rfc7807#page-6) , it seems as though it's possible to construct a ProblemDetails that doesn't round-trip when building an Api Client for `ApiController`.  I think that the next steps would be to further this so that "MVC on rails" common scenarios like "Entity Not Found" are handled better.

> > Do changes like this need to be mirrored in the aspnetcore repo? If they aren't, won't they be overwritten next time they're regenerated from C# XML comments?
> > -- @JamesNK 
> Is that what you and @wtgodbe did in https://github.com/dotnet/AspNetApiDocs/pull/187/files ?
> -- @Rick-Anderson 

This is by far the most confusing thing with Microsoft Docs.  The SQL Server team and @MightyPen seem to have nailed the process really well, but I get pretty confused when trying to contribute documentation to other Microsoft Docs pages.  That said, I do not know much about the internals of what the SQL Server Docs team did, and my (weak) understanding is Microsoft forked DocFX internally and created an undocumented hybrid version of DocFX where the community can't contribute to improve the extensions.

I've contributed a lot of PRs to fix Microsoft Docs for SQL Server, but I don't bother with other projects because the hurdle is too high to bother.

**Comment by Rick-Anderson** [comment]
@JamesNK  can I update the 6.0 version of the source, or do I need to update multiple versions?

**Comment by JamesNK** [comment]
We put new docs only into the master branch

**Comment by Rick-Anderson** [comment]
See https://github.com/dotnet/aspnetcore/pull/29292

**Comment by Rick-Anderson** [comment]
! I've contributed a lot of PRs to fix Microsoft Docs for SQL Server, but I don't bother with other projects because the hurdle is too high to bother.

@jzabroski try https://github.com/dotnet/AspNetCore.Docs, it's much simpler and I can help.

**Comment by jzabroski** [comment]
@Rick-Anderson Thanks for the tips.  Is Tenacious D going to start playing in the background if I ask why this repository is not as simple?

**Comment by Rick-Anderson** [comment]
@jzabroski XML docs have alot of rules that don't apply to most repos.

**Comment by KeterSCP** [comment]
@JamesNK, there is a typo in the link description as it says `Cookie policy middleware`. I tried to fix it in the https://github.com/dotnet/AspNetApiDocs/pull/231 but that PR was left hanging. What is the process of fixing it now? It still shows up with a typo: https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.mvc.apicontrollerattribute

**Pull request merged**